utcaApp/src/udriver.h

```diff
@@ -202,6 +212,15 @@ class UDriver: public asynPortDriver {
             return asynError;
         }
 
+        if (decoder_controller.count(function)) try {
+            if (first_general_parameter >= 0 && function <= last_general_parameter)
+                generic_decoder_controller->write_general(param_name, value);
+            else
+                generic_decoder_controller->write_channel(param_name, addr, value);
```

Comment:
We can add a call to `write_params` here to avoid having to define `writeInt32Impl` entirely for modules which only use these values.

It also allows us to eventually change the interface to return a token expressing the only register which has to be updated because of a change.

Reply:
> We can add a call to `write_params` here to avoid having to define `writeInt32Impl` entirely for modules which only use these values.

That seems nice. Can you see right away another module that would fall into this case?

> It also allows us to eventually change the interface to return a token expressing the only register which has to be updated because of a change.

When you say that, are you thinking about adding uhal support for single register update or should that already be possible?

Reply:
> That seems nice. Can you see right away another module that would fall into this case?

Ironically, with some of the changes being implemented by GIE, the RTMLAMP module will stop fitting into this case.

I haven't decided on whether I should rework things *now* to use the new model, or only whenever I change something in a module.

Either way, basically all of them fit into this case, because the `int32` writes will correspond directly to register fields. So only the other data types will require custom handling. And once I add fixed point conversion support, even more cases will be supported, once `writeDouble` is implemented by default...

For now, I think the only module for which I wouldn't use these functions is the acquisition one, because the class abstracts a lot of things away.

Reply:
> When you say that, are you thinking about adding uhal support for single register update or should that already be possible?

It would require adding support, it currently isn't possible. And I don't know if it's worth it, either. I have to figure out some way of testing write performance to figure out if it's a relevant bottleneck (especially considering these IOCs almost never get "fast" writes, only the FOFB ones during power supply cycling). However, if the writes take a long time, it's a long period without being able to read registers, so any writes that happen during critical operation periods --- which I don't think happen --- could also benefit from taking a shorter time.

Reply:
> testing write performance to figure out if it's a relevant bottleneck

That will probably be influenced by that bottleneck you already identified of having multiple PCI transfers for small amount of data, right?

>  it's a long period without being able to read registers

That's a good point. So far, you haven't needed to worry about that, am I right?

Reply:
> That's a good point. So far, you haven't needed to worry about that, am I right?

Exactly.